pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/JdbiMailAddress.sql.stg

```diff
@@ -95,6 +95,47 @@ findAllAddressesByParticipantGuid() ::= <<
             m.created_at
 
 >>
+
+findNonDefaultAddressesByParticipantIds(addressGuidSubstitutionName, participantIds) ::= <<
+        SELECT
+            u.guid AS participant_guid,
+            ais.activity_instance_id as instance_id,
+            m.address_id,
+            m.address_guid,
+            m.name,
+            m.street1,
+            m.street2,
+            m.city,
+            coalesce(s.code, m.state) AS state,
+            c.code AS country,
+            m.phone,
+            m.pluscode,
+            m.description,
+            vs.code AS validationStatus,
+            m.postal_code AS zip,
+            da.address_id AS defaultValue
+        FROM
+            mailing_address m
+               LEFT JOIN
+            country_address_info c ON c.country_address_info_id = m.country_id
+               LEFT JOIN
+            country_subnational_division s ON s.country_subnational_division_id = m.subnational_division_id
+               JOIN
+            user u on u.user_id=m.participant_user_id
+               LEFT JOIN
+            default_mailing_address da ON m.address_id=da.address_id
+               LEFT JOIN
+            mailing_address_validation_status vs on m.validation_status_id=vs.mailing_address_validation_status_id
+			   JOIN
+            activity_instance_substitution ais on ais.variable_name = :addressGuidSubstitutionName and ais.value = m.address_guid
```

Comment:
I'm curious about why this query need to look at activity substitutions? I'm sure we can write a more general query here to look for all addresses that are not marked as default?

Reply:
But how to get instanceIds? We need to group it by instances and get proper substitution for the currently exported instance.

Reply:
We have `Map<Long, Map<String, String>> activityInstanceSubstitutions` in `Participant`.

Reply:
Reimplemented according to your suggestion